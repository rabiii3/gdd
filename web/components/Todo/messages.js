import { defineMessages } from 'react-intl';

export default defineMessages({
  what: {
    id: 'components.Todo.what',
    defaultMessage: 'What',
  },
  who: {
    id: 'components.Todo.who',
    defaultMessage: 'Who',
  },
  when: {
    id: 'components.Todo.when',
    defaultMessage: 'When',
  },
});
