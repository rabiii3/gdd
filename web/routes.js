import { path, filter, compose, find, prop, reduce, toPairs } from 'ramda';

import PeoplePage from './pages/People/Loadable';
import UpdateResumePage from './pages/UpdateResume/Loadable';
import AddResumePage from './pages/AddResume/Loadable';
import RegisterAlmostTherePage from './pages/RegisterAlmostThere/Loadable';
import TenantsPage from './admin/Tenants/Loadable';
import ResetAlmostTherePage from './pages/ResetAlmostThere/Loadable';
import RequestConfirmationEmail from './pages/RequestConfirmationEmail/Loadable';
import RequestResetPasswordPage from './pages/RequestResetPassword/Loadable';
import ResetPasswordPage from './pages/ResetPassword/Loadable';
import ResumesPage from './pages/Resumes/Loadable';
import SkillsPage from './pages/Skills/Loadable';
import AgendaPage from './pages/Agenda/Loadable';
import StatisticsPage from './pages/Statistics/Loadable';
import PersonPage from './pages/Person/Loadable';
import Login from './components/Login/Loadable';
import UnauthorizedPage from './pages/Unauthorized/Loadable';
import NotFound from './pages/NotFound/Loadable';
import { ROLE } from '../lib/models/people';
import MODE from './utils/modes';
import AddTenantPage from './admin/AddTenant/Loadable';
import UpdateTenantPage from './admin/UpdateTenant/Loadable';
import LandingPage from './landing/Landing/Loadable';
import Jobs from './pages/Jobs/Loadable';
import AddJob from './pages/AddJob/Loadable';
import UpdateJob from './pages/UpdateJob/Loadable';
import TenantPage from './admin/Tenant/Loadable';

const routes = {
  resumes: {
    path: '/resumes',
    component: ResumesPage,
    exact: true,
    auth: true,
    default: true,
    mode: MODE.APP,
  },
  people: {
    path: '/people',
    component: PeoplePage,
    exact: true,
    auth: true,
    roles: [ROLE.admin, ROLE.headHunter],
    mode: MODE.APP,
  },
  person: {
    path: '/person/:id',
    component: PersonPage,
    exact: true,
    auth: true,
    mode: MODE.APP,
  },
  editResume: {
    path: '/resume/:id',
    component: UpdateResumePage,
    exact: true,
    auth: true,
    mode: MODE.APP,
  },
  addResume: {
    path: '/add_resume',
    component: AddResumePage,
    exact: true,
    auth: true,
    mode: MODE.APP,
  },
  signIn: {
    path: '/signIn',
    component: Login,
    exact: true,
    auth: false,
    default: true,
    mode: MODE.LOGIN,
  },

  reset_almost_there: {
    path: '/reset_almost_there',
    component: ResetAlmostTherePage,
    exact: true,
    auth: false,
    mode: MODE.LOGIN,
  },
  registerAlmostThere: {
    path: '/register_almost_there',
    component: RegisterAlmostTherePage,
    exact: true,
    auth: false,
    mode: MODE.LOGIN,
  },
  requestNewConfirmationEmail: {
    path: '/request_confirmation',
    component: RequestConfirmationEmail,
    exact: true,
    auth: false,
    mode: MODE.LOGIN,
  },
  requestResetPassword: {
    path: '/request_reset_password',
    component: RequestResetPasswordPage,
    exact: true,
    auth: false,
    mode: MODE.LOGIN,
  },
  resetPassword: {
    path: '/reset_password',
    component: ResetPasswordPage,
    exact: true,
    auth: false,
    mode: MODE.LOGIN,
  },
  unauthorized: {
    path: '/unauthorized',
    component: UnauthorizedPage,
    exact: true,
    auth: true,
    mode: MODE.ALL,
  },
  notFound: {
    path: '/not_found',
    component: NotFound,
    exact: true,
    auth: false,
    mode: MODE.ALL,
  },
  skills: {
    path: '/skills',
    component: SkillsPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin, ROLE.headHunter],
    mode: MODE.APP,
  },
  agenda: {
    path: '/agenda',
    component: AgendaPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin, ROLE.headHunter],
    mode: MODE.APP,
  },
  statistics: {
    path: '/statistics',
    component: StatisticsPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin, ROLE.headHunter],
    mode: MODE.APP,
  },
  tenants: {
    path: '/admin/tenants',
    component: TenantsPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin],
    default: true,
    mode: MODE.ADMIN,
  },
  addTenant: {
    path: '/admin/add_tenant',
    component: AddTenantPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin],
    default: true,
    mode: MODE.ADMIN,
  },
  updateTenant: {
    path: '/admin/update_tenant/:id',
    component: UpdateTenantPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin],
    default: true,
    mode: MODE.ADMIN,
  },
  viewTenant: {
    path: '/admin/tenant/:id',
    component: TenantPage,
    exact: true,
    auth: true,
    roles: [ROLE.admin],
    default: true,
    mode: MODE.ADMIN,
  },
  landing: {
    path: '/',
    component: LandingPage,
    default: true,
    mode: MODE.LANDING,
  },

  jobs: {
    path: '/jobs',
    component: Jobs,
    exact: true,
    auth: true,
    roles: [ROLE.admin, ROLE.headHunter],
    mode: MODE.APP,
  },
  addJob: {
    path: '/add_job',
    component: AddJob,
    exact: true,
    auth: true,
    mode: MODE.APP,
  },
  updateJob: {
    path: '/job/:id',
    component: UpdateJob,
    exact: true,
    auth: true,
    mode: MODE.APP,
  },
};

const exportedRoutes = compose(
  reduce((acc, [name, r]) => [...acc, { ...r, name }], []),
  toPairs,
  filter(prop('path')),
)(routes);

export const getDefaultRoute = (mode=MODE.APP) => find(route => route.default && route.mode === mode, exportedRoutes);
export const getRouteByName = name => routes[name];
export const getRouteByPath = path => find(r => r.path === path, exportedRoutes);
export const getPathByName = (name, param) => {
  const path = prop('path', getRouteByName(name));
  return param ? `${path.replace(':id', param)}` : path;
};
export const getRouteProp = prop => name => path([name, prop], routes);
export const getRouteRoles = getRouteProp('roles');
export const getRouteAuthProps = name => ({ roles: getRouteRoles(name) });
export const isAuthRequired = route => route && (route.auth || route.roles);
export const signInRoute = '/signIn';
// export const makeSignInRoute = (url, tenant) => `${url}${signInRoute}?tenant=${tenant}&_p=${btoa(JSON.stringify({ mode: MODE.LOGIN }))}`;
export const makeSignInRoute = (url, tenant) => `${url}${signInRoute}?tenant=${tenant}`;
export const isSignInRoute = ({ pathname }) => pathname === signInRoute;

export default {
  getRoutes: () => exportedRoutes,
  getDefaultRoute,
  getRouteByName,
  getPathByName,
  getRouteByPath,
  getRouteProp,
  getRouteRoles,
  getRouteAuthProps,
  isAuthRequired,
  signInRoute,
  makeSignInRoute,
  isSignInRoute,
};
