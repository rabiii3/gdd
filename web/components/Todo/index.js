import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'react-final-form';
import { compose, map, propOr, propEq } from 'ramda';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import classNames from 'classnames';
import connect from 'react-redux/lib/connect/connect';
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';
import MenuItem from '@material-ui/core/MenuItem/MenuItem';
import { withStyles } from '@material-ui/core/styles';
import Call from '@material-ui/icons/Call';
import Forward from '@material-ui/icons/Forward';
import Face from '@material-ui/icons/Face';
import Block from '@material-ui/icons/Block';
import Grid from '@material-ui/core/Grid';
import messages from './messages';
import DateTimePickerForm from '../../forms/DateTimePicker';
import withRoutes from '../../hoc/routes';
import { SMALL } from '../../widgets/ResumeStatus';
import Select from '../../forms/Select';
import MultiSelectUsers from '../../forms/MultiSelectUsers';
import { getAdminsAndHeadHunters, getPeople } from '../../selectors/people';
import { Spacer } from '../../widgets/Spacer';

const style = theme => ({
  marginDatePicker: {
    marginTop: '1%'
  },
  marginWho: {
    marginTop: '2%'
  },
  marginTop: {
    [theme.breakpoints.up('lg')]: {
      marginTop: '5%'
    },
    marginTop: '8%'
  },

  marginTopWhen: {
    marginTop: '3%'
  },
  noMarginTop: {
    [theme.breakpoints.up('lg')]: {
      marginTop: '-5%'
    },
    marginTop: '-8%'
  },
  noMarginTopWhen: {
    marginTop: '1%'
  },
  labelEdit: {
    color: '#0000008a',
    fontSize: '12px',
    fontFamily: 'Roboto',

  },
});

export const Todo = ({ classes, people, comment, adminsOrHeadHunters, width, fullWidth, values }) => {

  const reasons = [
    { id: 'none', icon: <Block />, label: 'None', value: 'none' },
    { id: 'call', icon: <Call />, label: 'Call', value: 'call' },
    { id: 'forward', icon: <Forward />, label: 'Forward', value: 'forward' },
    { id: 'meet', icon: <Face />, label: 'Meet', value: 'meet' },
  ];

  const whoDefaultValue = comment =>
    compose(
      map(element => ({ value: element, label: people[element] })),
      propOr([], 'who')
    )(comment);

  const sendtoWhom = whoDefaultValue(comment);
  return (

    <Grid container direction="row" justify="space-between" alignitem="center">
      <Grid item xs={12} lg={fullWidth ? 12 : 4}>
        <Field
          className={width}
          name="what"
          label={<label > <FormattedMessage {...messages.what} /> </label >}
          component={Select}
        >
          {map(reason => (
            <MenuItem key={reason.id} value={reason.value}>
              <Grid container direction="row" justify="flex-start" alignitem="center">
                {reason.icon}
                <Spacer size={SMALL} />
                {reason.label}
              </Grid>
            </MenuItem>
          ))(reasons)}
        </Field>
      </Grid>
      
      <Grid item xs={12} lg={fullWidth ? 12 : 4}>
        { !propEq('what', 'forward', values) &&(
        <Grid container direction="column" justify="flex-end" alignItems="flex-start" className={!values.when ? classes.marginTopWhen : classes.noMarginTopWhen}>
          <Field
            component={DateTimePickerForm}
            name="when"
            className={classNames(width, fullWidth && classes.marginDatePicker)}
            label={values.when && <FormattedMessage {...messages.when} />}
          />
        </Grid>)}
      </Grid>
      
      <Grid item xs={12} lg={fullWidth ? 12 : 4}  >
        <Grid container direction="column" justify="flex-end" alignItems="flex-start" className={(!values.who || values.who && values.who[0] === undefined) ? classes.marginTop : classes.npMarginTop}>
          <Grid item>
            {values.who && values.who[0] && <label className={classes.labelEdit}> <FormattedMessage {...messages.who} /></label>}
          </Grid>
          <Grid item >
            <Field
              className={classNames(width, fullWidth && classes.marginWho)}
              component={MultiSelectUsers}
              name="who"
              type="select"
              placeholder={<FormattedMessage {...messages.who} />}
              isMulti
              defaultValue={propEq('what', 'none', values) ? [] : sendtoWhom}
              suggestion={adminsOrHeadHunters}
              textFieldProps={{
                InputLabelProps: {
                  shrink: true,
                },
              }}
            />
          </Grid>
        </Grid>
      </Grid>
    </Grid>

  );
};

Todo.propTypes = {
  classes: PropTypes.object,
  adminsOrHeadHunters: PropTypes.array,
  width: PropTypes.string,
  fullWidth: PropTypes.bool,
  values: PropTypes.object,
  people: PropTypes.object,
  comment: PropTypes.object,
};

const mapStateToProps = () =>
  createStructuredSelector({
    adminsOrHeadHunters: getAdminsAndHeadHunters,
    people: getPeople,
  });

export const enhance = compose(
  withStyles(style),
  connect(
    mapStateToProps,
    null,
  ),
  withRoutes,
);
export default enhance(Todo);
