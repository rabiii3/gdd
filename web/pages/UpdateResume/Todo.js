import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedRelative } from 'react-intl';
import classNames from 'classnames';
import { prop, propEq, compose, map, isNil } from 'ramda';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Call from '@material-ui/icons/Call';
import Forward from '@material-ui/icons/Forward';
import Face from '@material-ui/icons/Face';
import { Spacer, UserName } from '../../widgets';
import { UserAvatar } from '../../widgets/Avatar';
import { SMALL } from '../../widgets/ResumeStatus';
import messages from './messages';
import withRoutes from '../../hoc/routes';
import { eventToLabel } from '../../../lib/models/comments';
import { isTodoCall, isTodoForward, isTodoMeet } from '../../../lib/models/todo';


const style = theme => ({
  grey: {
    color: 'grey',
  },
  comment: {
    paddingTop: theme.spacing.unit * 1,
    paddingBottom: theme.spacing.unit * 1,
    paddingLeft: theme.spacing.unit * 1,
    display: 'flex',
    flexDirection: 'column',
  },
  small: {
    fontSize: 'small',
  },
  smaller: {
    fontSize: 'x-small',
  },
  bold: {
    fontWeight: 'bold',
  },content: { 
    marginTop: '-8px', 
    marginBottom: '-20px', 
    fontWeight: '300',
    margin: theme.spacing.unit,
  },
  margin: {
    margin: theme.spacing.unit,
  },
  iconFontSize: { 
    fontSize: 22 
  },
});

export const Todo = ({ classes, todo, comment, people, routes, isShownInComment, isResumePage }) => (
  
  todo ? 
  (
    <Grid item>
      { comment && ( isShownInComment  ||  ((new Date(prop('when',comment)) >= Date.now()) || isNil(prop('when', comment))) ) &&(
        <Grid container direction={propEq('forward','what',comment) ? 'row-reverse' : 'row'} justify="center" alignItems="center" spacing={1} wrap="nowrap">
          <Grid item>
            <Grid container direction="row">
              { !(prop('who', comment)) ?
                ( (!isResumePage && prop('what',comment)) &&
                  (
                    <Grid item key={prop('createdBy',comment)} className={classes.margin}>
                      <UserAvatar
                        classes={classes.avatar}
                        user={people[prop('createdBy',comment)]}
                        size="XSMALL"
                      />
                    </Grid>
                  )
                )
                :
                (
                  map(user => 
                  <Grid item key={user} className={classes.margin}>
                    <UserAvatar
                      classes={classes.avatar}
                      user={people[user]}
                      size="XSMALL"
                    />
                  </Grid>)(prop('who',comment))
                )
              }   
            </Grid>
          </Grid>
          <Grid item>
            {isTodoCall(comment) && <Call color="action" className={classes.iconFontSize} />}
            {isTodoForward(comment) && <Forward color="action" className={classes.iconFontSize} />}
            {isTodoMeet(comment) && <Face color="action" className={classes.iconFontSize} />}
          </Grid>
          <Grid item>
            <Typography className={classNames(classes.grey, classes.when, classes.smaller)} variant="caption" noWrap>
              {prop('when',comment) ? <FormattedRelative value={prop('when', comment)} /> : ''}
            </Typography>
          </Grid>       
        </Grid>
        )
      }
    </Grid>
  )
  :
  (
    <Grid container direction="column"> 
      {(comment && !isShownInComment) && (
      <Grid item container alignItems="center">
        <Grid item>
          <UserAvatar
            classes={classes.avatar}
            user={people[prop('createdBy',comment)]}
            size={SMALL}
            to={people[prop('createdBy',comment)] && routes.getPathByName('person',prop('_id', people[prop('createdBy',comment)]))}
          />
        </Grid>
        <Grid item xs className={classNames(classes.margin, classes.comment)}>
          <Grid container direction="row" justify="flex-start" alignItems="center">
            <Typography
              className={classNames(classes.grey, classes.small)}
              variant="subtitle1"
            >
              {eventToLabel(prop('event',comment))}
              {(prop('event', comment)) && <FormattedMessage {...messages.by} />}
            </Typography>
            <Spacer size={SMALL} />
              <Grid className={classNames(classes.grey, classes.small, classes.bold)}>
                <UserName user={people[prop('createdBy',comment)]} />   
              </Grid>
            <Spacer size={SMALL} />
            <Typography className={classNames(classes.grey, classes.smaller)} variant="caption" align="right">
              <FormattedRelative value={prop('createdAt',comment)} />
            </Typography>
          </Grid>
          <Grid item>
            <Typography className={classes.content} dangerouslySetInnerHTML={{ __html: comment.content }} />
          </Grid>
        </Grid>
      </Grid>
      )}
    </Grid>
  )
);

Todo.propTypes = {
  classes: PropTypes.object,
  todo: PropTypes.bool,
  comment: PropTypes.object,
  routes: PropTypes.object,
  people: PropTypes.object,
  isShownInComment: PropTypes.bool,
  isResumePage: PropTypes.bool,
}

const enhance = compose(
  withStyles(style),
  withRoutes,
);

export default enhance(Todo);
