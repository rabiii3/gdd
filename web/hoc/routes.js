import React from 'react';
import { RoutesContextConsumer } from '../components/RoutesContext';

const withRoutes = Component => {
  const EnhancedComponent = props => (
    <RoutesContextConsumer>
      {routes => <Component {...props} routes={routes} />}
    </RoutesContextConsumer>
  );
  return EnhancedComponent;
};

export default withRoutes;
