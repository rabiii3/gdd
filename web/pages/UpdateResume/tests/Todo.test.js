import React from 'react';
import { shallow } from 'enzyme';
import { Todo } from '../Todo';
import routes from '../../../routes';

describe('web | pages | UpdateResume | Todo', () => {
  describe('render', () => {
    it('should match snapshot', () => {
      const props = {
        classes: {},
        people: {},
        todo: true,
        comment: {
          who:[]
        },
        routes,
        isShownInComment: false, 
        isResumePage:false,
      };
      const wrapper = shallow(<Todo {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
