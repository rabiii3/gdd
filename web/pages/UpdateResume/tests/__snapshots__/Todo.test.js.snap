// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web | pages | UpdateResume | Todo render should match snapshot 1`] = `
<WithStyles(Grid)
  item={true}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    direction="row"
    justify="flex-start"
    spacing={8}
    wrap="nowrap"
  >
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Grid)
        container={true}
        direction="row"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    />
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Typography)
        className=""
        noWrap={true}
        variant="caption"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
